ts/multicolor_series.ts

```diff
@@ -0,0 +1,729 @@
+import SeriesRegistry from 'highcharts/ts/Core/Series/SeriesRegistry';
+import Utilities from 'highcharts/ts/Core/Utilities';
+import LineSeries from 'highcharts/ts/Series/Line/LineSeries';
+import type PointerEvent from 'highcharts/ts/Core/PointerEvent';
+import type SVGElement from 'highcharts/ts/Core/Renderer/SVG/SVGElement';
+import type SVGAttributes from 'highcharts/ts/Core/Renderer/SVG/SVGAttributes';
+import type ColorType from 'highcharts/ts/Core/Color/ColorType';
+import type SVGPath from 'highcharts/ts/Core/Renderer/SVG/SVGPath';
+import { type StatesOptionsKey } from 'highcharts/ts/Core/Series/StatesOptions';
+import {
+    type ColoredlineSeriesOptions,
+    type SeriesColoredGraphPath,
+    type SeriesColoredPoint,
+    type SeriesColoredSegment,
+    type SeriesColoredSegmentPath
+} from 'types';
+
+const { area: AreaSeries } = SeriesRegistry.seriesTypes;
+const { extend, isArray } = Utilities;
+
+/**
+ *
+ *  Type guards
+ *
+ */
+
+const isSVGPathSegment =
+    (value: SeriesColoredSegmentPath[]): value is SVGPath.Segment[] => true;
+
+/**
+ *
+ *  Functions
+ *
+ */
+
+function getPath (
+    graphPaths: SeriesColoredGraphPath[]
+): SeriesColoredSegmentPath[] {
+    let segmentPath: SeriesColoredSegmentPath[] = [];
+
+    if (graphPaths) {
+        graphPaths.forEach(function (el): void {
+            if (isArray(segmentPath) && isArray(el[0])) {
+                segmentPath = segmentPath.concat(el[0]);
+            }
+        });
+    }
+
+    return segmentPath;
+}
+
+/**
+ *
+ * @private
+ * @class
+ * @name Highcharts.seriesTypes.coloredline
+ *
+ */
+
+class ColoredlineSeries extends LineSeries {
+    
+    /**
+     *
+     *  Constructor
+     *
+     */
+
+    constructor () {
+        super();
+
+        this.singlePoints = [];
+        this.graphPaths = [];
+        this.areaPaths = [];
+        this.graphs = [];
+    }
+
+    /**
+     *
+     *  Properties
+     *
+     */
+
+    public pointRange: number | undefined;
+
+    public singlePoints!: SeriesColoredPoint[];
+
+    public points!: SeriesColoredPoint[];
+
+    public segments!: SeriesColoredSegment[];
+
+    public options!: ColoredlineSeriesOptions;
```

Comment:
It's used to prevent defining these variables in the constructor. Of course, we need a default value for some in the constructor, but I think it's not necessary for all. What do you think?

Reply:
I think we should avoid using it like that. Try to remove it and initialize what is needed in the constructor, perhaps not all types are needed?

Reply:
I removed the `!` check for all variables apart from the `points`. Having it defined initially breaks the chart. The logic comes from the `translate` method, and I guess that at some point, HC checks if this variable is defined, so perhaps there'll be a need to rebuild some extra HC methods.